src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -246,12 +246,41 @@ class ConstraintsChecker
         return true;
     }
 
-    template <typename T, typename U, std::enable_if_t<std::is_enum<T>::value && !std::is_pointer<U>::value, int> = 0>
+    template <typename T, typename U,
+              std::enable_if_t<std::is_enum<T>::value && !std::is_enum<U>::value && !std::is_pointer<U>::value, int> = 0>
     bool CheckConstraintMinValue(const char * itemName, T current, U expected)
     {
         return CheckConstraintMinValue(itemName, chip::to_underlying(current), expected);
     }
 
+    template <typename T, typename U,
+              std::enable_if_t<std::is_enum<T>::value && std::is_enum<U>::value && std::is_convertible<U, int>::value, int> = 0>
+    bool CheckConstraintMinValue(const char * itemName, T current, U expected)
+    {
+        return CheckConstraintMinValue(itemName, chip::to_underlying(current), chip::to_underlying(expected));
+    }
+
+    template <typename T, typename U,
+              std::enable_if_t<std::is_enum<T>::value && std::is_enum<U>::value && !std::is_convertible<U, int>::value, int> = 0>
+    bool CheckConstraintMinValue(const char * itemName, T current, U expected)
+    {
+        return CheckConstraintMinValue(itemName, chip::to_underlying(current), chip::to_underlying(expected));
+    }
+
+    template <typename T, typename U,
+              std::enable_if_t<!std::is_enum<T>::value && std::is_enum<U>::value && std::is_convertible<U, int>::value, int> = 0>
```

Comment:
Again, I don't see why we need separate cases for `std::is_convertible<U, int>::value` and `!std::is_convertible<U, int>::value`.